[web] Move web-only initialization APIs to `dart:ui_web`

mdebbar opened the pull request:
| Old API in `dart:ui` | New API in `dart:ui_web` |
|-|-|
| ~`webOnlyInitializePlatform`~ | ~`ui_web.initializePlatform`~ |
| `webOnlyWarmupEngine` | `ui_web.bootstrapEngine` |
| `debugEmulateFlutterTesterEnvironment` | `ui_web.debugEmulateFlutterTesterEnvironment` |
| `webOnlySetPluginHandler` | `ui_web.setPluginHandler` |

Part of https://github.com/flutter/flutter/issues/126831

yjbanov commented on lib/web_ui/lib/ui_web/src/ui_web/initialization.dart:
```diff
@@ -0,0 +1,66 @@
+/// methods (via [AppBootstrap.autoStart]).
+///
+/// This method should NOT trigger the download of any additional resources
+/// (except when the app is in "autoStart" mode).
```
Maybe we should move most of this under a `//` doc and write a fresh one that doesn't rely on internal details?

yjbanov commented on lib/web_ui/lib/ui_web/src/ui_web/initialization.dart:
```diff
@@ -0,0 +1,66 @@
+/// (except when the app is in "autoStart" mode).
+Future<void> warmupEngine({
+  Function? registerPlugins,
+  Function? runApp,
```
Can we use more specific function signatures? Looks like they are just `void Function()`?

ditman commented on lib/web_ui/lib/initialization.dart:
```diff
@@ -23,94 +23,66 @@
-  }
+// TODO(mdebbar): Deprecate this and remove it.
+// https://github.com/flutter/flutter/issues/127395
+bool get debugEmulateFlutterTesterEnvironment {
```
Is this something that is needed for end users, or is this only for our testing? If it's only for our engine testing, it probably should live within `engine` or similar, rather than be exposed to `web_ui`?

ditman replied:
(Never mind, this is used all over the bootstrap and the framework!)

yjbanov replied:
Alternatively, this could be hidden behind JS-interop, if we don't want it to show up in the public APIs.

mdebbar replied:
`debugEmulateFlutterTesterEnvironment` has effect on several things including the usage of Ahem font. If users want their tests to produce real text (e.g. for screenshots), they have to disable `debugEmulateFlutterTesterEnvironment`.

mdebbar commented on lib/web_ui/lib/ui_web/src/ui_web/initialization.dart:
```diff
@@ -0,0 +1,66 @@
+/// Flutter framework.
+///
+/// Unlike [warmupEngine], this method initializes UI and non-UI services.
+Future<void> initializePlatform() async {
```
No reason. I just removed the `webOnly` prefix from the name. I think `initializeEngine` makes more sense, and is more consistent with `warmupEngine`.

mdebbar commented on lib/web_ui/lib/ui_web/src/ui_web/initialization.dart:
```diff
@@ -0,0 +1,66 @@
+/// Initializes essential bits of the engine before it fully initializes.
+///
+/// When [FlutterLoaderExtension.didCreateEngineInitializer] is set, it delegates
+/// engine initialization and app startup to the programmer. Else, it immediately
```
@ditman any ideas here?

Most people won't see or need this. It's done by `flutter.js` for them. The people who care about this are the ones who want to provide their own replacement for `flutter.js`.

mdebbar commented:
1. Not moving `webOnlyInitializePlatform` to `dart:ui_web` anymore. We might be able to switch all call sites to use `bootstrapEngine` instead.
2. Renamed `warmupEngine` to `bootstrapEngine`.
3. Updated the docs of `bootstrapEngine`.

yjbanov commented on lib/web_ui/lib/ui_web/src/ui_web/initialization.dart:
```diff
@@ -0,0 +1,57 @@
+/// The [runApp] callback is invoked to run the app after the engine is fully
+/// initialized.
+///
+/// For more information, see what the `flutter_tools` doesin the entrypoint
```
nit: s/doesin/does in/

yjbanov commented on lib/web_ui/lib/ui_web/src/ui_web/plugins.dart:
```diff
@@ -0,0 +1,22 @@
+import 'package:ui/src/engine.dart';
+import 'package:ui/ui.dart' as ui;
+
+typedef PluginMessageHandler = Future<void> Function(
```
Does this need a dartdoc?

mdebbar replied:
I removed this. We already have the same typedef in `dart:ui` so I used that instead.
